src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor

```diff
@@ -28,56 +54,91 @@
 
 @code {
     private bool expanded = true;
+    private bool expandedGroup = false;
+
+    private IEnumerable<Channel> SelectedChannels = Array.Empty<Channel>();
+    private IImmutableList<Channel> availableChannels = ImmutableList<Channel>.Empty;
+
+    private List<Category> Categories = new List<Category>();
 
-    class Category(string name)
+    public class Category
     {
-        public string Name { get; set; } = name;
-        public List<Channel> Channels { get; } = [];
-    }
+        public string Name { get; set; }
+        public List<Channel> Channels { get; } = new List<Channel>();
 
-    private List<Category> Categories =
-    [
-        new Category(".NET 10"),
-        new Category(".NET 9"),
-        new Category(".NET 8"),
-        new Category(".NET 6"),
-        new Category(".NET"),
-        new Category("VS"),
-        new Category("Windows"),
-    ];
+        public Category(string name)
+        {
+            Name = name;
+        }
+    }
 
     protected override async Task OnInitializedAsync()
     {
-        var channels = await PcsApi.Channels.ListChannelsAsync();
-        var otherCategory = new Category("Other");
+        availableChannels = await PcsApi.Channels.ListChannelsAsync();
+        Categories = ConstructCategories(availableChannels);
+    }
+
+    private List<Category> ConstructCategories(IEnumerable<Channel> channels)
+    {
+        var categories = new List<Category>
+        {
+            new Category(".NET 10"),
+            new Category(".NET 9"),
+            new Category(".NET 8"),
+            new Category(".NET 6"),
+            new Category(".NET"),
+            new Category("VS"),
+            new Category("Windows"),
+        };
 
+        var otherCategory = new Category("Other");
         foreach (var channel in channels)
         {
             bool categorized = false;
-            foreach (var category in Categories)
+            foreach (var category in categories)
             {
                 if (channel.Name.StartsWith(category.Name))
```

Comment:
Can we add a category named "Test" and all categories with Classification equal "test" would go there? As a top priority rule?

Reply:
Sure thing, on it